draft-ion.adoc

```diff
@@ -208,79 +200,176 @@ represented as follows:
+
+All Ion links have one or more <<links, link relation types>>.  A link's total set of link relation types is the set union of the link's implicit link relation type and the link's explicit link relation types.
+
+==== Implicit Link Relation Type
+
+All Ion links have exactly one implicit link relation type.  The implicit link relation type can be one of three possible values depending on where the link is located in the Ion content:
+
+1. Root Object
+2. Named Link
+3. Collection Resource `items` array element
+
+===== Root Object Implicit Link Relation Type
+
+If the Root Object is also found to be an Ion Link, it has an implicit link relation type of `self`.
+
+A non-normative example of a Root Object that is also an Ion Link:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users/1" },
+  "name": "Joe"
+}
+----
 
-=== Registered Metadata Field Names
+===== Named Link Implicit Link Relation Type
 
-The following field names are registered in the IANA Ion Metadata field registry
-defined in [TBD].  None of the fields defined below are intended to be mandatory
+If a JSON Object member value is an Ion Link, the Ion Link is implicitly assigned a link relation type that equals the member name.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "employer": { "meta": { "href": "https://api.example.com/corporations/acme" } }
+}
+----
+
+The above example shows an Ion Link member named `employer`.  The octet sequence `employer` is therefore assigned as the link's implicit link relation type.
+
+===== Collection Item Implicit Link Relation Type
+
+Any Ion Link in a Collection Resource's `items` array is implicitly assigned the {uri-linkRelType-item}[`item`] link relation type.
+
+A non-normative example:
+
+[source,json]
+----
+{
+  "meta": { "href": "https://api.example.com/users" },
+  "items": [
+    { "meta": { "href": "https://api.example.com/users/1" } },
+    { "meta": { "href": "https://api.example.com/users/2" } }
+  ]
+}
+----
+
+The above example has 2 Ion Links in its `items` array.  Each of these 2 links are implicitly assigned the {uri-linkRelType-item}[`item`] link relation type.
+
+==== Explicit Link Relation Types
+
+An Ion link MAY have one or more explicitly defined link relation types declared as an array of strings assigned to the link Meta Object's `rel` member.
+
+A non-normative example of a link with an explicitly defined link relation type:
+
+.Example Ion Link with explicit link relation type:
+[source,json]
+----
+{ "meta": { "href": "https://cdn.example.com/example.ico", "rel": ["icon"] } }
+----
+
+The above example has an explicit link relation type of http://www.w3.org/TR/html5/links.html#rel-icon[`icon`].
+
+More than one explicit link relation type MAY be defined in the `rel` array.
+
+Ion parsers MUST NOT recognize any non-string value in the `rel` array as a link relation type.
+
+Ion parsers MUST treat null or empty `rel` values as if the `rel` property was not defined at all.
```

Comment:
This happens to with member links.

```json
{
  "employer": { 
    "meta": {
      "rel": ["company"],
      "href": "https://api.example.com/corporations/acme"
    }
}
```

Would the link relations be both employer and company? Is this something to specify in the spec?